### PR TITLE
Added ToDoItem sample Web API and .NET client

### PR DESCRIPTION
**Pull request by jasonjoh**
The sample API is intended to give developers a simple API that they can generate clients for with Kiota.

The .NET client is a simple console app that uses a Kiota-generated SDK to call the sample API.

### CODE REVIEW

**Comment by jasonjoh** [comment]
@baywet this is the sample API we talked about yesterday.

File: sample-api/api/src/Controllers/ToDoItemsController.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            await LogRequestAsync(Request);
+            _context.ToDoItems.Add(todoItem);
+            await _context.SaveChangesAsync();
+
+            return Created(todoItem);
+            //return CreatedAtAction(nameof(GetToDoItem), new { id = todoItem.Id }, todoItem);
+        }
+
+        // DELETE: /ToDoItems/5
+        //[HttpDelete("{id}")]
```
Comment (baywet) [inline sample-api/api/src/Controllers/ToDoItemsController.cs:110]:
```suggestion
```

File: sample-api/api/src/Controllers/ToDoItemsController.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        [HttpDelete]
+        public async Task<IActionResult> DeleteToDoItemAsync(Guid key)
```
Comment (baywet) [inline sample-api/api/src/Controllers/ToDoItemsController.cs:112]: from URL? or from patH?

File: sample-api/clients/dotnet/src/Program.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+},
+settings.TenantId,
+settings.ClientId);
+
+var context = new TokenRequestContext(scopes);
+var response = await credential.GetTokenAsync(context);
+
+Console.WriteLine($"Token: {response.Token}");
+
+var authProvider = new AzureIdentityAuthenticationProvider(credential, scopes);
```
Comment (baywet) [inline sample-api/clients/dotnet/src/Program.cs:33]: once https://github.com/microsoft/kiota/pull/1051 gets merged, you might want to pass in the allowed hosts so it keeps working.

File: sample-api/clients/dotnet/src/ToDoClient.csproj [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    <None Update="appsettings.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
+    <None Update="appsettings.local.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Identity" Version="1.5.0" />
```
Comment (baywet) [inline sample-api/clients/dotnet/src/ToDoClient.csproj:20]: no nuget.config to configure the packages sources?

File: sample-api/api/src/Controllers/ToDoItemsController.cs (hunk 2 above)
Comment (jasonjoh) [reply sample-api/api/src/Controllers/ToDoItemsController.cs:106]: It's in the path: `DELETE /ToDoItems/{guid}`

Comment (baywet) [reply sample-api/api/src/Controllers/ToDoItemsController.cs:106]: ok I though an attribute on the parameter was required, but apparently it maps the parameters automatically.
Can you remove the commented out attribute please?

File: sample-api/clients/dotnet/src/ToDoClient.csproj (hunk 4 above)
Comment (jasonjoh) [reply sample-api/clients/dotnet/src/ToDoClient.csproj:20]: Added along with note in README

File: sample-api/api/src/Controllers/OpenApiController.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        public ActionResult GetOpenApiDoc()
+        {
+            var odataBuilder = new ODataConventionModelBuilder();
+            odataBuilder.EnableLowerCamelCase();
+            odataBuilder.EntitySet<ToDoItem>("ToDoItems");
+            var model = odataBuilder.GetEdmModel();
+
+            var openApi = model.ConvertToOpenApi(new OpenApiConvertSettings
+            {
+                ServiceRoot = new Uri("https://localhost:7206/"),
```
[the remark this answers is not in the document]
Comment (jasonjoh) [reply sample-api/api/src/Controllers/OpenApiController.cs]: Fixed

**Review by baywet — APPROVED** [review]

**Comment by jasonjoh** [comment]
Thanks @baywet. I don't have the ability to merge in this repo. Could you merge this?

**Pull request merged**